src/dsl/validation.rs

```diff
@@ -116,11 +158,58 @@ mod tests {
               - name: timeout
                 expect: facts.corosync_token_timeout == values.expected_token_timeout
         "#;
+
+        let engine = Engine::new();
+
         let json_value: serde_json::Value =
             serde_yaml::from_str(&input).expect("Unable to parse yaml");
         let json_schema = get_json_schema();
-        let validation_result = validate(&json_value, "156F64", &json_schema);
+        let validation_result = validate(&json_value, "156F64", &json_schema, &engine);
 
         assert_eq!(validation_result.is_ok(), true);
     }
+
+    #[test]
+    fn validate_invalid_expectations_check() {
+        let input = r#"
```

Comment:
Maybe we could remove most of this boilerplate test data, only leaving the needed fields. In this case only the expectations and values part are important.
Just to reduce the number of lines hehe
But well, if you want to have all the data for consistency is fine too